lib/irb.rb

```diff
@@ -1102,24 +1105,52 @@ def each_top_level_statement
       end
     end
 
+    COMMAND_FLAG_REGEXP = /(?<cmd_flag>-[a-zA-Z]+( +\S+)*)/
+    COMMAND_ARG_REGEXP = /(?<cmd_arg>[^- ]\S*)/
+    COMMAND_NAME_REGEXP = /(?<cmd_name>\S+)/
+    SIMPLE_COMMAND_REGEXP = /^#{COMMAND_NAME_REGEXP}\z/
+    COMMAND_WITH_ARGS_REGEXP = /^#{COMMAND_NAME_REGEXP} +#{COMMAND_ARG_REGEXP}\z/
+    COMMAND_WITH_FLAGS_REGEXP = /^#{COMMAND_NAME_REGEXP} +#{COMMAND_FLAG_REGEXP}\z/
+    COMMAND_WITH_ARGS_AND_FLAGS_REGEXP = /^#{COMMAND_NAME_REGEXP} +#{COMMAND_ARG_REGEXP} +#{COMMAND_FLAG_REGEXP}\z/
```

Comment:
I think this still matches to the second line. It should be `/^` → `/\A`

Command (Last line matches to `COMMAND_REGEXP`)
```ruby
show_source = 1..
2# Set#include?
```
Not a command (Both line does not match to `COMMAND_REGEXP`)
```ruby
show_source = 1..
2 # Set#include?
```